destination/destination_integration_test.go

```diff
@@ -0,0 +1,395 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/conduitio-labs/conduit-connector-mongo/config"
+	"github.com/conduitio-labs/conduit-connector-mongo/destination/writer"
+)
+
+const (
+	// set the directConnection to true in order to avoid the known hostname problem.
+	testURI              = "mongodb://localhost:27017/?directConnection=true"
+	testDB               = "test"
+	testCollectionPrefix = "test_coll"
+
+	// next consts will be used for test models.
+	testIDFieldName = "_id"
+	testID          = "6384c3e48740eb54f858bde5"
+
+	testTextFieldName = "textfield"
+	testTextData      = "some text data"
+
+	testNumberFieldName  = "numberfield"
+	testNumberData       = 1234
+	testNumberDataChange = 5678
+)
+
+func TestDestination_Write_snapshotSuccess(t *testing.T) {
+	is := is.New(t)
+
+	cfg := prepareConfig(t)
+
+	destination := NewDestination()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	col, err := getTestCollection(ctx, cfg[config.KeyCollection])
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = col.Drop(context.Background())
+		is.NoErr(err)
+
+		err = destination.Teardown(ctx)
+		is.NoErr(err)
+	})
+
+	n, err := destination.Write(ctx, []sdk.Record{getTestSnapshot(t)})
+	is.NoErr(err)
+	is.Equal(n, 1)
+
+	c, err := col.CountDocuments(ctx, bson.D{})
+	is.NoErr(err)
+	is.Equal(c, int64(1))
+
+	res, err := col.Find(ctx, bson.M{})
+	res.Next(ctx)
+	var result map[string]any
+	err = res.Decode(&result)
+	is.NoErr(err)
+
+	id, ok := result[testIDFieldName].(primitive.ObjectID)
+	is.Equal(ok, true)
+
+	hex, err := primitive.ObjectIDFromHex(testID)
+	is.NoErr(err)
+
+	is.Equal(id, hex)
+
+	text, ok := result[testTextFieldName].(string)
+	is.Equal(ok, true)
+	is.Equal(text, testTextData)
+
+	num, ok := result[testNumberFieldName].(float64)
+	is.Equal(ok, true)
+	is.Equal(int(num), testNumberData)
```

Comment:
It seems this can be a function, like this:
```go
func compareTestPayload(
	t *testing.T,
	ctx context.Context,
	is *is.I,
	col *mongo.Collection,
	testRecordPayload sdk.StructuredData,
) {
	t.Helper()

	c, err := col.CountDocuments(ctx, bson.D{})
	is.NoErr(err)
	is.Equal(c, int64(1))

	res, err := col.Find(ctx, bson.M{})
	is.NoErr(err)
	is.True(res.TryNext(ctx))
	var result map[string]any
	err = res.Decode(&result)
	is.NoErr(err)

	is.Equal(sdk.StructuredData(result), testRecordPayload)
}
```

Reply:
added